src/main/java/edu/ucla/library/iiif/fester/utils/ThumbnailUtils.java

```diff
@@ -0,0 +1,80 @@
+
+package edu.ucla.library.iiif.fester.utils;
+
+import edu.ucla.library.iiif.fester.Constants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Utility methods for thumbnail selection
+*/
+public final class ThumbnailUtils {
+    private static final String HEADER_THUMB = "thumbnail";
+
+    private ThumbnailUtils() {
+    }
+
+   /**
+     * Adds a "thumbnail" column to a CSV if needed..
+     *
+     * @param aCsvList A CSV parsed into a list of arrays
+     * @return The CSV with added "thumbnail" column
+   */
+    public static List<String[]> addThumbnailColumn(final List<String[]> aCsvList) {
+        Objects.requireNonNull(aCsvList);
+        final List<String[]> csvList;
+
+        if (hasThumbnailColumn(aCsvList.get(0))) {
+            return aCsvList;
+        } else {
+            final int newArraySize = aCsvList.get(0).length + 1;
+            csvList = new ArrayList<>(aCsvList.size());
+            aCsvList.stream().forEach(entry -> csvList.add(Arrays.copyOf(entry, newArraySize)));
+            csvList.get(0)[newArraySize - 1] = HEADER_THUMB;
+            return csvList;
+        }
+    }
+
+   /**
+     * Adds a base IIF URL for the thumnail image.
+     *
+     * @param aCsvList A CSV in list of arrays format
+     * @param aURL The thumbnail URL
+   */
+    public static void addThumbnailURL(final List<String[]> aCsvList, final String aURL) {
+        Objects.requireNonNull(aCsvList);
+        final int thumbnailIndex = Arrays.asList(aCsvList.get(0)).indexOf(HEADER_THUMB);
+        for (int index = 1; index < aCsvList.size(); index++ ) {
```

Comment:
The control flow logic here and in the caller appears incorrect. The loop over the rows should be somewhere in the caller of this method, with each row getting a different thumbnail URL, instead of each row (work) being given the same thumbnail URL